src/libmongoc/tests/test-mongoc-mongohouse.c

```diff
@@ -165,8 +166,9 @@ cmd_succeeded_cb (const mongoc_apm_command_succeeded_t *event)
       BSON_ASSERT (BSON_ITER_HOLDS_UTF8 (&child_iter));
 
       temp = bson_iter_utf8 (&child_iter, NULL);
-      test->cursor_ns = (char *) bson_malloc0 (strlen (temp));
-      strncpy (test->cursor_ns, temp, strlen (temp));
+      temp_size = 1 + strlen(temp);
+      test->cursor_ns = (char *) bson_malloc0 (temp_size);
```

Comment:
This can be simplified with [bson_strdup](http://mongoc.org/libbson/current/bson_strdup.html):

```
test->cursor_ns = bson_strdup (temp);
```

Reply:
+1 bson_strdup() makes this much prettier! I'll update the commit with the version number (this is indeed a newer compiler, 9.3).